v2/tools/generator/internal/astmodel/primitive_type.go

```diff
@@ -47,6 +47,24 @@ var ErrorType = &PrimitiveType{"error", "nil"}
 // assert that we implemented Type correctly
 var _ Type = (*PrimitiveType)(nil)
 
+var cachedPrimitives = map[string]*PrimitiveType{
+	"int":         IntType,
+	"uint64":      UInt64Type,
+	"uint32":      UInt32Type,
+	"string":      StringType,
+	"armid":       ARMIDType,
+	"float64":     FloatType,
+	"bool":        BoolType,
+	"interface{}": AnyType,
+	"any":         AnyType,
+	"error":       ErrorType,
+}
+
+func LookupPrimitiveType(name string) (*PrimitiveType, bool) {
```

Comment:
minor: Consider adding a comment here explaining what the purpose of the method is?

Presumably it shouldn't really be used in a "Go code" context, because there you could just go `astmodel.IntType`, which is shorter than `astmodel.LookupPrimitiveType("int")` (and handling the bool response). The real usage is for contexts where you don't have access to the symbols (like YAML).

Reply:
Added comment.